runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java

```diff
@@ -270,6 +271,57 @@ public void onChatMessage(ChatMessage event)
 			{
 				timer = new RaidsTimer(getRaidsIcon(), this, Instant.now());
 				infoBoxManager.addInfoBox(timer);
+
+				if (config.layoutChatMsg())
+				{
+					// Write the layout, and then every room, the same way the panel
+					// overlay displays it. The layout, then combat & puzzle rooms.
+					String layout = getRaid().getLayout().toCode().replaceAll("#", "").replaceAll("¤", "");
+					ChatMessageBuilder msgBuilder = new ChatMessageBuilder()
+							.append(ChatColorType.HIGHLIGHT)
+							.append("[")
+							.append(layout)
+							.append("]: ");
+
+					boolean firstElement = true;
+					for (Room layoutRoom : getRaid().getLayout().getRooms())
```

Comment:
I would personally use streams for this as it makes the code quite a bit easier to read:
```
getRaid().getLayout().getRooms().stream()
			.map(e -> getRaid().getRoom(e.getPosition()))
			.filter(Objects::nonNull)
			.filter(e -> e.getType() == RaidRoom.Type.COMBAT || e.getType() == RaidRoom.Type.PUZZLE)
			.map(e -> e.getType() == RaidRoom.Type.COMBAT ? e.getBoss().getName() : e.getPuzzle().getName())
			.reduce((s1, s2) -> s1 + ", " + s2)
			.ifPresent(msgBuilder::append);
```